core/internal/gqlmock/gqlmock.go

```diff
@@ -71,6 +71,24 @@ func (c *MockClient) StubMatchOnce(
 		})
 }
 
+// StubMatchWithError registers a response for a matching request.
+//
+// The next time a request matching `requestMatcher` is made, the
+// `err` will be returned.
+func (c *MockClient) StubMatchWithError(
```

Comment:
maybe better signature is resp + error?

Reply:
This matches the function for graphql `MakeRequest` which only returns `error`
The response data is set in the response parameter (like [here](https://github.com/wandb/wandb/blob/main/core/internal/gqlmock/gqlmock.go#L85)). 

If we want to add the response as well later we can copy in copy a similar pattern above.

Reply:
no this is perfect, thanks!